internal/incident/incidents.go

```diff
@@ -69,37 +155,14 @@ func GetCurrent(
 	created := false
 	currentIncident := currentIncidents[obj]
 
-	if currentIncident == nil {
-		ir := &IncidentRow{}
-		incidentLogger := logger.With(zap.String("object", obj.DisplayName()))
-		incident := NewIncident(db, obj, runtimeConfig, incidentLogger)
+	if currentIncident == nil && create {
+		created = true
 
-		err := db.QueryRowxContext(ctx, db.Rebind(db.BuildSelectStmt(ir, ir)+` WHERE "object_id" = ? AND "recovered_at" IS NULL`), obj.ID).StructScan(ir)
-		if err != nil && !errors.Is(err, sql.ErrNoRows) {
-			logger.Errorw("Failed to load incident from database", zap.String("object", obj.DisplayName()), zap.Error(err))
-
-			return nil, false, errors.New("failed to load incident from database")
-		} else if err == nil {
-			incident.incidentRowID = ir.ID
-			incident.StartedAt = ir.StartedAt.Time()
-			incident.Severity = ir.Severity
-			incident.logger = logger.With(zap.String("object", obj.DisplayName()), zap.String("incident", incident.String()))
-
-			if err := incident.restoreEscalationsState(ctx); err != nil {
-				return nil, false, err
-			}
-
-			currentIncident = incident
-		}
-
-		if create && currentIncident == nil {
-			created = true
-			currentIncident = incident
-		}
+		incidentLogger := logger.With(zap.String("object", obj.DisplayName()))
+		currentIncident = NewIncident(db, obj, runtimeConfig, incidentLogger)
+		currentIncident.ObjectID = obj.ID
```

Comment:
If the object is already passed to `NewIncident()` already, that function should take care of initializing the `ObjectID` field.